pkg/objstore/oss/oss.go

```diff
@@ -0,0 +1,310 @@
+	if err := b.bucket.DeleteObject(name); err != nil {
+		return errors.Wrap(err, "delete oss object")
+	}
+	return nil
+}
+
+func configFromEnv() Config {
+	c := Config{
+		EndPoint:  os.Getenv("OSS_ENDPOINT"),
+		Bucket:    os.Getenv("OSS_BUCKET"),
+		SecretKey: os.Getenv("OSS_SECRET_KEY"),
+		SecretId:  os.Getenv("OSS_SECRET_ID"),
+	}
+
+	return c
+}
+
+// ValidateForTests checks to see the config options for tests are set.
+func ValidateForTests(conf Config) error {
+	if conf.EndPoint == "" ||
+		conf.SecretId == "" ||
+		conf.SecretKey == "" {
+		return errors.New("insufficient oss test configuration information")
+	}
+	return nil
+}
+
+func parseConfig(conf []byte) (Config, error) {
+	var config Config
+	if err := yaml.Unmarshal(conf, &config); err != nil {
+		return Config{}, err
+	}
+
+	return config, nil
+}
+
+func NewBucket(logger log.Logger, conf []byte, component string) (*Bucket, error) {
+	config, err := parseConfig(conf)
+	if err != nil {
+		return nil, err
+	}
+	if err := validate(config); err != nil {
+		return nil, err
+	}
+
+	client, err := alioss.New(config.EndPoint, config.SecretId, config.SecretKey)
+	if err != nil {
+		return nil, err
+	}
+	bk, err := client.Bucket(config.Bucket)
+	if err != nil {
+		return nil, errors.Wrap(err, "oss bucket")
+	}
+
+	bkt := &Bucket{
+		logger: logger,
+		client: client,
+		name:   config.Bucket,
+		config: config,
+		bucket: bk,
+	}
+	return bkt, nil
+}
+
+// Iter calls f for each entry in the given directory (not recursive.). The argument to f is the full
+// object name including the prefix of the inspected directory.
+func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error) error {
+	if dir != "" {
+		dir = strings.TrimSuffix(dir, objstore.DirDelim) + objstore.DirDelim
+	}
+
+	marker := alioss.Marker("")
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		objects, err2 := b.bucket.ListObjects(alioss.Prefix(dir), alioss.Delimiter(objstore.DirDelim), marker)
+		if err2 != nil {
+			return errors.Wrap(err2, "oss bucket list")
+		}
+		for _, object := range objects.Objects {
+
+			if object.Key == "" {
+				continue
+			}
+			if err := f(object.Key); err != nil {
+				return err
+			}
+		}
+		for _, object := range objects.CommonPrefixes {
+			if err := f(object); err != nil {
+				return err
+			}
+		}
+		marker = alioss.Marker(objects.NextMarker)
+		if !objects.IsTruncated {
+			break
+		}
+	}
+
+	return nil
+}
+
+func (b *Bucket) Name() string {
+	return b.name
+}
+
+// validate checks to see the config options are set.
+func validate(conf Config) error {
+	if conf.EndPoint == "" {
+		return errors.New("no oss endpoint in config file")
+	}
+	if conf.Bucket == "" {
+		return errors.New("no oss bucket in config file")
+	}
+
+	if conf.SecretId == "" && conf.SecretKey != "" {
+		return errors.New("no oss acccess_key specified while secret_key is present in config file; both of them should be present in config.")
+	}
```

Comment:
Names(access_key or secret_key or secret) should keep consistent with documentation.

Reply:
OK